tests/test_churn_classification/test_churn_classification.py

```diff
@@ -58,12 +58,17 @@ def __run(self):
             submission = tp.submit(scenario)
             if self.waiting_jobs_to_complete:
                 # 12 jobs must be processed to complete the scenario. It may take some time.
-                assert_true_after_time(
-                    lambda: submission.submission_status == SubmissionStatus.COMPLETED,
-                    time=120,
-                    msg=lambda s: f"Submission status is {s.submission_status} after 120 seconds",
-                    s=submission,
-                )
+                try:
+                    assert_true_after_time(
+                        lambda: submission.submission_status == SubmissionStatus.COMPLETED,
+                        time=120,
+                        msg=lambda s: f"Submission status is {s.submission_status} after 120 seconds",
+                        s=submission,
+                    )
+                except Exception:
+                    print("Failed with submission status:", submission.submission_status)
+                    print("Failed with job statuses:", {job.id: job.status for job in submission.jobs})
```

Comment:
The msg parameter of the `assert_true_after_time` function accepts a lambda function to print a message in case of failure.
I believe instead of try catching the assert, we need to change the lambda function.